miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java

```diff
@@ -56,18 +57,32 @@
 import uk.ac.bbsrc.tgac.miso.core.data.type.HealthType;
 import uk.ac.bbsrc.tgac.miso.core.data.type.PlatformType;
 import uk.ac.bbsrc.tgac.miso.core.security.util.LimsSecurityUtils;
+import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
 import uk.ac.bbsrc.tgac.miso.service.ChangeLogService;
 import uk.ac.bbsrc.tgac.miso.service.PlatformService;
 import uk.ac.bbsrc.tgac.miso.service.SequencerReferenceService;
 import uk.ac.bbsrc.tgac.miso.service.SequencingParametersService;
 import uk.ac.bbsrc.tgac.miso.service.impl.RunService;
+import uk.ac.bbsrc.tgac.miso.webapp.util.JsonArrayCollector;
+import uk.ac.bbsrc.tgac.miso.webapp.util.RunMetricsSource;
 
 @Controller
 @RequestMapping("/run")
 @SessionAttributes("run")
 public class EditRunController {
   protected static final Logger log = LoggerFactory.getLogger(EditRunController.class);
 
+  /**
+   * Get a stream of source of metrics
+   * 
+   * Normally, metrics collected by run scanner are stored in the MISO database, but it is possible to provide
+   * 
+   * @return
+   */
+  public Stream<RunMetricsSource> getSources() {
+    return Stream.of(Run::getMetrics);
```

Comment:
I think I'm missing something here... `getMetrics` is an instance method on `Run`, which returns the run's metrcs as a `String`. How is this getting "sources," rather than just the metrics, and where is it getting them from?